### PR TITLE
setup.py: auto install pkgconfig and check depends

### PR DESCRIPTION
**Pull request by zoumingzhe**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -13,11 +13,28 @@
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library; if not, see <http://www.gnu.org/licenses/>.
 
-import pkgconfig
+try:
+    import pkgconfig
+except Exception:
+    from pip import _internal as pip_internal
+    pip_internal.main(['install', 'pkgconfig'])
+    import pkgconfig
```
Comment (vojtechtrefny) [inline setup.py:21]: I am not sure if I like the idea of automatically installing something with `pip` here, users might be confused why running `python3 setup.py ...` installs a package with `pip`. I'll try to see if there is a way to make `pip` install `pkgconfig` by itself (`install_requires=["pkgconfig"]` is obviously not enough for build time dependencies). Maybe having a `requirements.txt` would work better or possibly switching to `pyproject.toml` which has some way to [specify build dependencies](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/#build-time-dependencies). I'll think about it and let you now as soon as possible.

**Comment by vojtechtrefny** [comment]
I experimented a little with pyproject.toml and #5 is intended to replace this -- including pkgconfig in `build-system.requires` seems to be enough for `pip` to pull it in when installing from PyPI.

File: setup.py (hunk 1 above)
Comment (zoumingzhe) [reply setup.py:21]: It is the best solution to make install by itself. However, the package's name may not be consistent for different distributions.

**Comment by vojtechtrefny** [comment]
Replaced by #5 which is now merged.

**Pull request closed without merging**